js/src/ValidateSPV.js

```diff
@@ -1,232 +1,253 @@
+    digest, version, prevHash, merkleRoot, timestamp, target, nonce
+  };
+}
 
-      // Require that the header has sufficient work
-      digest = btcUtils.hash256(header);
-      if (!module.exports.validateHeaderWork(btcUtils.reverseEndianness(digest), target)) {
-        throw new Error('Header does not meet its own difficulty target.');
-      }
+/**
+ *
+ * Checks validity of header chain
+ *
+ * @dev                   Compares the hash of each header to the prevHash in the next header
+ * @param {Uint8Array}    headers Raw byte array of header chain
+ * @returns {BigInt}      The total accumulated difficulty of the header chain, or an error code
+ * @throws {TypeError}    When passed a chain that contains junk data
+ * @throws {Error}        When passed an invalid chain, or  header with llow work
+*/
+export function validateHeaderChain(headers) {
+  // Check header chain length
+  if (headers.length % 80 !== 0) {
+    throw new TypeError('Header bytes not multiple of 80.');
+  }
 
-      totalDifficulty += btcUtils.calculateDifficulty(target);
-    }
-    return totalDifficulty;
-  },
-
-  /**
-   * @notice                Checks validity of header work
-   * @param {Uint8Array}    digest Header digest
-   * @param {Uint8Array}    target The target threshold
-   * @returns {Boolean}     True if header work is valid, false otherwise
-   */
-  validateHeaderWork: (digest, target) => {
-    if (digest === 0) {
-      return false;
+  // Initialize header start index
+  let digest;
+  let totalDifficulty = BigInt(0);
+
+  for (let i = 0; i < headers.length / 80; i += 1) {
+    // ith header start index and ith header
+    const start = i * 80;
+    const header = utils.safeSlice(headers, start, start + 80);
+
+    // After the first header, check that headers are in a chain
+    if (i !== 0) {
+      if (!module.exports.validateHeaderPrevHash(header, digest)) {
+        throw new Error('Header bytes not a valid chain.');
+      }
     }
-    return utils.bytesToUint(digest) < target;
-  },
-
-  /**
-   * @notice                Checks validity of header chain
-   * @dev                   Compares current header prevHash to previous header's digest
-   * @param {Uint8Array}    header The raw bytes header
-   * @param {Uint8Array}    prevHeaderDigest The previous header's digest
-   * @returns {Boolean}     True if header chain is valid, false otherwise
-   */
-  validateHeaderPrevHash: (header, prevHeaderDigest) => {
-    // Extract prevHash of current header
-    const prevHash = btcUtils.extractPrevBlockLE(header);
-
-    // Compare prevHash of current header to previous header's digest
-    if (!utils.typedArraysAreEqual(prevHash, prevHeaderDigest)) {
-      return false;
+
+    // ith header target
+    const target = BTCUtils.extractTarget(header);
+
+    // Require that the header has sufficient work
+    digest = BTCUtils.hash256(header);
+    if (!module.exports.validateHeaderWork(BTCUtils.reverseEndianness(digest), target)) {
```

Comment:
don't need 'module.exports' as long as the function is defined above, so `validateHeaderWork` needs to be moved.